src/setup_payload/tests/TestQRCode.cpp

```diff
@@ -460,6 +460,8 @@ TEST(TestQRCode, TestExtractPayload)
     EXPECT_EQ(QRCodeSetupPayloadParser::ExtractPayload(std::string("%Z%MT:ABC")), std::string("ABC"));
     EXPECT_EQ(QRCodeSetupPayloadParser::ExtractPayload(std::string("%Z%MT:ABC%")), std::string("ABC"));
     EXPECT_EQ(QRCodeSetupPayloadParser::ExtractPayload(std::string("%Z%MT:ABC%DDD")), std::string("ABC"));
+    EXPECT_EQ(QRCodeSetupPayloadParser::ExtractPayload(std::string("%Z%MT:ABC*DEF*GHI%DDD")), std::string("ABC*DEF*GHI"));
+    EXPECT_EQ(QRCodeSetupPayloadParser::ExtractPayload(std::string("%Z*WX%MT:ABC*DEF*GHI%DDD*EEE")), std::string("ABC*DEF*GHI"));
```

Comment:
Why not balk at this bad code, instead?

Reply:
ExtractPayload is just existing code that looks for "thing between % starting with MT:".  It does no parsing/validation of the payload.....  I didn't change its behavior at all, just added tests for the new setting it might be used in.